_posts/2020-12-06-converting-your-laptop-to-a-janus-modem-using-unetaudio.md

```diff
@@ -0,0 +1,227 @@
+
+Next, you can make a JANUS transmission by typing:
+
+```
+> phy << new TxJanusFrameReq()
+AGREE
+phy >> TxFrameNtf:INFORM[type:#3 txTime:127730682]
+```
+
+This will transmit a broadcast JANUS frame.
+
+On the receiving computer, you should see:
+
+```
+phy >> RxFrameStartNtf:INFORM[type:#3 rxTime:127730375 rxDuration:1100000 detector:0.96]
+phy >> RxJanusFrameNtf:INFORM[type:#3 classUserID:0 appType:0 appData:0 mobility:false canForward:true txRxFlag:true rxTime:127730374 rssi:-44.2 cfo:0.0]
+```
+
+> NOTE: The JANUS frame format doesn’t contain user data by default, and so the MTU is 0 (`frameLength` of 8 as defined in JANUS specifications). JANUS allows vendors to provide their own “cargo”, and so UnetStack lets you increase the `frameLength` to make space for user data (`MTU`).
+
+
+If you would like to add a payload, you can increase the `phy.frameLength` parameter (default is 8). Let us transmit 1 byte of data.
+
+```
+> phy[3].frameLength=9
+9
+```
+
+Data can be added to the transmission by typing:
+```
+> phy << new TxJanusFrameReq(data:[1])
+AGREE
+phy >> TxFrameNtf:INFORM[type:#3 txTime:1754760016]
+```
+
+On the receiving side:
+
+```
+> phy << new TxJanusFrameReq(data:[1])
+AGREE
+phy >> RxFrameStartNtf:INFORM[type:#3 rxTime:1754759709 rxDuration:1300000 detector:0.96]
+phy >> RxJanusFrameNtf:INFORM[type:#3 classUserID:0 appType:0 appData:0 mobility:false canForward:true txRxFlag:true reservationDuration:0.15031052 rxTime:1754759708 rssi:-44.3 cfo:0.0 (1 byte)]
+```
+
+
+You can check the content of the received frame by typing:
+```
+> ntf.data
+[1]
+```
+
+It is that simple to set up an SODAM on your laptop/computer and transmit and receive acoustic communication signals using Unet audio.
+
+
+## Conclusion
+
+Unet audio provides an easy method for researchers, teachers and students to convert their computers to a JANUS compliant SDOAM. Even the cheapest of the underwater acoustic modems comes at a cost and may not be the best option to invest in, if the intended purpose is teaching. Unet audio community edition is free for academic and research use.
+
+Even if you happen to own some UnetStack compatible acoustic modems, testing with a modem means setting up a dedicated test setup & equipment, access to a water body and dealing with the logistics. With Unet audio, you can develop and test from the comfort of your seat. Additionally, Unet audio provides an SDOAM that covers the frequency band of operation as defined by the JANUS specifications.
+
+This post only gives you a glimpse of how you can do JANUS transmissions and receptions using Unet audio. There are many more interesting things you can do like transmit and receive the raw baseband signals from a `*.wav` file and plot a spectrogram to verify the signal is indeed JANUS compliant ([as illustrated here](https://stackoverflow.com/questions/65027940/how-do-i-record-janus-signal-as-wav-file)), or setup a network of SDOAMs that uses JANUS as their physical layer.
+
```

Comment:
Does it make sense to also link to the video of the presentation from Oceans2019 at John's workshop? I believe it's on our YouTube channel.

Reply:
It is only the highlights video, which I have added to the text.